From 015668134ce3cbc6c3e2c65f6c55ab36810b787e Mon Sep 17 00:00:00 2001
From: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Date: Fri, 21 Jun 2024 08:44:27 +0200
Subject: [PATCH 637/954] fix(stm32mp2): maintain LPSRAM1 content in standby
 mode

This fix allow to maintain LPSRAM1 RAM content in Standy mode in
particular when used by Cortex-CM0.
This fix is not valid for STM3MP21 as it doesn't embed a Cortex-CM0

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 34bf558e5270f681d7d0873645406e9a892fe11c fix(stm32mp2): maintain LPSRAM1 content in standby mode]
Change-Id: If9f0a70769123b2254336bff3235b18d2f31f1b9
Signed-off-by: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/388925
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_pm.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_pm.c b/plat/st/stm32mp2/stm32mp2_pm.c
index 6f29c7d25b..b6d59a87b6 100644
--- a/plat/st/stm32mp2/stm32mp2_pm.c
+++ b/plat/st/stm32mp2/stm32mp2_pm.c
@@ -1065,8 +1065,13 @@ static void stm32_pm_init(void *fdt)
 
 	mmio_write_32(rcc_base + RCC_C1SREQCLRR, RCC_C1SREQSETR_STPREQ_MASK);
 
+#if STM32MP21
 	/* Maintain BKPSRAM & RETRAM content in Standby */
 	mmio_write_32(pwr_base + PWR_CR9, PWR_CR9_BKPRBSEN);
+#else
+	/* Maintain BKPSRAM & LPSRAM1 & RETRAM content in Standby */
+	mmio_write_32(pwr_base + PWR_CR9, PWR_CR9_BKPRBSEN|PWR_CR9_LPR1BSEN);
+#endif
 	mmio_write_32(pwr_base + PWR_CR10, PWR_CR10_RETRBSEN_STANDBY);
 
 	/* Prevent RETRAM erase */
-- 
2.53.0

